### PR TITLE
Multiple template engines docs

### PR DESCRIPTION
**Pull request by aaugustin**

### CODE REVIEW

**Comment by timgraham** [comment]
I hope to review this tomorrow. If you have a chance to remove the code changes you already committed, that would be great.

**Comment by aaugustin** [comment]
@timgraham done. This is still missing some documentation for `django.template.Engine`.

File: docs/topics/templates.txt [hunk 1]
```diff
@@ -1,704 +1,587 @@
...
 
-.. seealso::
+    If context is provided, it must be a :class:`dict`. If it isn't provided,
+    the engine will render the template with an empty context.
 
-    Django's admin interface can include a complete reference of all template
-    tags and filters available for a given site. See
-    :doc:`/ref/contrib/admin/admindocs`.
+    If request is provided, it must be a :class:`~django.http.HttpRequest`.
+    Then the engine must make it available in the template as well as the CSRF
```
Comment (carljm) [inline docs/topics/templates.txt]: "in the template context"?

**Comment by carljm** [comment]
I gave this a fairly quick read-through, and didn't see anything obviously problematic. I didn't review thoroughly.

File: docs/topics/templates.txt (hunk 1 above)
Comment (aaugustin) [inline docs/topics/templates.txt]: Fixed.

**Comment by aaugustin** [comment]
Here's the remainder of the documentation (blocker for the 1.8 alpha release).

@timgraham @carljm I didn't modify the commits you already reviewed. I'll squash later.  c97575f rewrites some parts of 4b3a38f.

File: docs/topics/templates.txt [hunk 2]
```diff
@@ -1,704 +1,645 @@
...
+.. admonition:: About this section
+
+    This is an overview of the Django template language's APIs. For details
+    see the :doc:`API reference </ref/templates/api>`.
+
+Engine
+~~~~~~
+
+:class:`django.template.Engine` encapsulates an instance of the Django
+template system. The main reason for instanciating an
```
Comment (timgraham) [inline docs/topics/templates.txt]: instantiating

File: docs/ref/templates/api.txt [hunk 3]
```diff
@@ -508,18 +617,13 @@ optional, third positional argument, ``processors``. In this example, the
...
-    ``RequestContext``). To use a ``RequestContext`` in your template
-    rendering, use the :meth:`~django.shortcuts.render()` shortcut which is
-    the same as a call to :func:`~django.shortcuts.render_to_response()` with a
-    ``context_instance`` argument that forces the use of a ``RequestContext``.
+Built-in template context processors
+------------------------------------
 
 Here's what each of the default processors does:
 
+.. _context-processors:
```
Comment (timgraham) [inline docs/ref/templates/api.txt]: Please move this above the "Built-in template context processors"

File: docs/topics/templates.txt (hunk 2 above)
Comment (aaugustin) [inline docs/topics/templates.txt]: fixed

**Pull request merged**